carbonserver/render.go

```diff
@@ -90,8 +88,6 @@ func getFormat(req *http.Request) (responseFormat, error) {
 }
 
 func getTargets(req *http.Request, format responseFormat) (map[timeRange][]target, error) {
-	req.ParseForm() // idempotent
```

Comment:
I think here we need to have more clever logic, because there might be multiple `target=` fields.

Reply:
It actually works just fine (see below). Our first call to `req.FormValue()` to get `from` and `until` will call `req.ParseMultipartForm()`, itself calling `req.ParseForm` as necessary. Therefore, `req.Form` is filled when we start iterating over `eq.Form["target"]`

## Content-Type: application/x-www-form-urlencoded

```
$ curl -vv -d "format=pickle;local=1;noCache=1;from=1536704479;until=1536715279;now=1536715279;target=carbon.agents.graphite-cache-0.cache.maxSize&target=carbon.agents.graphite-cache-1.cache.maxSize" "localhost:8080/render/"

```
Found target: carbon.agents.graphite-cache-0.cache.maxSize
Found target: carbon.agents.graphite-cache-1.cache.maxSize

[2018-09-12T17:25:37.544-0700] INFO [access] fetch served {"handler": "render", "url": "/render/", "peer": "[::1]:54108", "targets": ["carbon.agents.graphite-cache-0.cache.maxSize", "carbon.agents.graphite-cache-1.cache.maxSize"], "format": "pickle", "runtime_seconds": 0.000221681, "query_cache_enabled": true, "from_cache": false, "metrics_fetched": 0, "values_fetched": 0, "memory_used_bytes": 0, "http_code": 200}
```

## Content-Type: multipart/form-data

```
$ curl -vv -F "format=pickle" -F "local=1" -F "noCache=1" -F "from=1536704479" -F "until=1536715279" -F "now=1536715279" -F "target=carbon.agents.graphite-cache-0.cache.maxSize" -F "target=carbon.agents.graphite-cache-1.cache.maxSize" "localhost:8080/render/"
```

```
Found target: carbon.agents.graphite-cache-0.cache.maxSize
Found target: carbon.agents.graphite-cache-1.cache.maxSize
[2018-09-12T17:26:05.073-0700] INFO [access] fetch served {"handler": "render", "url": "/render/", "peer": "[::1]:54113", "targets": ["carbon.agents.graphite-cache-0.cache.maxSize", "carbon.agents.graphite-cache-1.cache.maxSize"], "format": "pickle", "runtime_seconds": 0.000284352, "query_cache_enabled": true, "from_cache": true, "metrics_fetched": 0, "values_fetched": 0, "memory_used_bytes": 0, "http_code": 200}
```